keyboards/rocketboard_16/rules.mk

```diff
@@ -1,8 +1,19 @@
 # MCU name
 MCU = STM32F103
 
-# Bootloader selection
-BOOTLOADER = stm32duino
+# # Bootloader selection
+# BOOTLOADER = stm32duino
+
+# Custom loader configuration
+MCU_LDSCRIPT = STM32F103C8T6_custom
+OPT_DEFS += -DBOOTLOADER_STM32DUINO
+BOARD = STM32_F103_STM32DUINO
+STM32_BOOTLOADER_ADDRESS = 0x80000000
+DFU_ARGS = -d 1EAF:0003 -a 2 -R
+DFU_SUFFIX_ARGS = -v 1EAF -p 0003
```

Comment:
```suggestion
# Bootloader selection
BOOTLOADER = stm32duino

# Custom loader configuration
MCU_LDSCRIPT = STM32F103C8T6_custom
```

Reply:
This change completely undoes the solution. See @sigprof's explanation [here](https://github.com/qmk/qmk_firmware/pull/12537#discussion_r625119378l). Tested with your suggested changes and it no longer links due to size issue. With @sigprof's code it links just fine.

Reply:
Maybe worth putting a comment in the makefile about the issue.

Reply:
Alternatively, given the particular issue here, `MCU_LDSCRIPT` could be changed to a `?=` in bootloader.mk, so that my changes do work (and potentially a second ldscript in `platforms/chibios/common/ld` for 128kB chips).

Reply:
So... is this good to merge then?

Reply:
Bump